intersection_transit_maneuvering/src/intersection_transit_maneuvering.cpp

```diff
@@ -0,0 +1,233 @@
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+#include <ros/ros.h>
+#include <string>
+#include <algorithm>
+#include <memory>
+#include <boost/uuid/uuid_generators.hpp>
+#include <boost/uuid/uuid_io.hpp>
+#include <lanelet2_core/geometry/Point.h>
+#include <trajectory_utils/trajectory_utils.h>
+#include <trajectory_utils/conversions/conversions.h>
+#include <sstream>
+#include <carma_utils/containers/containers.h>
+#include <Eigen/Core>
+#include <Eigen/Geometry>
+#include <Eigen/LU>
+#include <Eigen/SVD>
+#include <intersection_transit_maneuvering.h>
+
+using oss = std::ostringstream;
+
+namespace intersection_transit_maneuvering
+{
+    IntersectionTransitManeuvering::IntersectionTransitManeuvering(carma_wm::WorldModelConstPtr wm)
+    {        
+        plugin_discovery_msg_.name = "IntersectionTransitManeuvering";
+        plugin_discovery_msg_.versionId = "v1.0";
+        plugin_discovery_msg_.available = true;
+        plugin_discovery_msg_.activated = false;
+        plugin_discovery_msg_.type = cav_msgs::Plugin::TACTICAL;
+        plugin_discovery_msg_.capability = "tactical_plan/plan_trajectory";
+    }
+
+    bool IntersectionTransitManeuvering::plan_trajectory_cb(cav_srvs::PlanTrajectoryRequest& req, cav_srvs::PlanTrajectoryResponse& resp)
+    {
+        
+        ros::WallTime start_time = ros::WallTime::now();  // Start timeing the execution time for planning so it can be logged
+
+        lanelet::BasicPoint2d veh_pos(req.vehicle_state.X_pos_global, req.vehicle_state.Y_pos_global);
+        double current_downtrack = wm_->routeTrackPos(veh_pos).downtrack;
+
+        std::vector<cav_msgs::Maneuver> maneuver_plan;
+        for(size_t i = req.maneuver_index_to_plan; i < req.maneuver_plan.maneuvers.size(); i++)
+        {
+            if(req.maneuver_plan.maneuvers[i].type == cav_msgs::Maneuver::INTERSECTION_TRANSIT_STRAIGHT ||
+            req.maneuver_plan.maneuvers[i].type == cav_msgs::Maneuver::INTERSECTION_TRANSIT_LEFT || 
+            req.maneuver_plan.maneuvers[i].type == cav_msgs::Maneuver::INTERSECTION_TRANSIT_RIGHT )
+            {
+                maneuver_plan.push_back(req.maneuver_plan.maneuvers[i]);
+                resp.related_maneuvers.push_back(i);
+            }
+            else
+                {
+                    break;
+                }
+        }
+        cav_srvs::PlanTrajectoryRequest& req2;
+        auto new_plan = convert_maneuver_plan(maneuver_plan);
+
+        for(const auto& man : new_plan)
+            {
+                req2.maneuver_plan.push_back(man);
+            }
+
+        /*Once the maneuvers have been successfully converted, call the inlanecruising plugin to calculate the trajectory*/
+
+        if (trajectory_client_ && trajectory_client_.exists() && trajectory_client_.isValid())
```

Comment:
This logic cannot be unit tested because it requires an active ros connection. Looks like this is based on the yield call from inlane-cruising which as we discussed is not a good approach. You want to follow the pattern used for publishers in that class. This means using dependency injection, you want to pass in an object to your IntersectionTransitManeuver constructor that could provide a call() method. There are two common ways to do this, either use std::function which matches the ros::Service::call api or define a c++ interface with a call method that you can create different implementations for.